### ISSUE TITLE
Zero-cost all the syntax.

### ISSUE THREAD

**Issue by aloiscochard** [scalaz/scalaz#1229]
We use macro to make our syntax calls more efficien, for example:

- usage https://github.com/scalaz/scalaz/blob/series/8.0.x/base/src/main/scala/typeclass/FunctorSyntax.scala#L1-- macro def https://github.com/scalaz/scalaz/blob/series/8.0.x/meta/src/main/scala/Ops.scala#L9-L13 

We should now do that for all the current syntax, which should hopefully give enough `Ops` combinators to make it easy to contribute new one.

**Comment by wedens** [comment]
More cases are covered in [machinist](https://github.com/typelevel/machinist/blob/master/src/main/scala/spire/ops/Ops.scala). Not sure if we need all of them though.

**Comment by aloiscochard** [comment]
@wedens I don't think we will, and that was the main motivation to copy the code instead of depending on the lib.

We'll see along the way!

**Comment by S11001001** [comment]
As the code was copied, we should be sure to include its license (at the point of copying) as part of the Scalaz 8 license.  This should be straightforward enough. #1334
